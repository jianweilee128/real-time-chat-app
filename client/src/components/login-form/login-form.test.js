import React from "react";
import { shallow } from "enzyme";

import LoginForm from "./login-form.component";

describe("LoginForm component", () => {
  let wrapper;
  let mockLogin;

  beforeEach(() => {
    mockLogin = jest.fn();

    const mockProps = {
      login: mockLogin,
    };
    wrapper = shallow(<LoginForm.WrappedComponent {...mockProps} />);
  });

  it("should render LoginForm component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call login function when login button is clicked", () => {
    wrapper.find(".login-btn").simulate("click");
    expect(mockLogin).toHaveBeenCalled();
  });
});
