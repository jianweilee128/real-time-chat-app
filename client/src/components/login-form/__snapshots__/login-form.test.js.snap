// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component should render LoginForm component 1`] = `ShallowWrapper {}`;
